src/openslide-decode-jxr.c

```diff
@@ -0,0 +1,260 @@
+/*
+ *  OpenSlide, a library for reading whole slide image files
+ *
+ *  Copyright (c) 2007-2015 Carnegie Mellon University
+ *  Copyright (c) 2011 Google, Inc.
+ *  Copyright (c) 2015 Benjamin Gilbert
+ *  All rights reserved.
+ *
+ *  OpenSlide is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, version 2.1.
+ *
+ *  OpenSlide is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with OpenSlide. If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <string.h>
+#include <config.h>
+
+#include "openslide-private.h"
+#include "openslide-decode-jxr.h"
+
+#define BGR24TOARGB32(p)                                                       \
+  (0xFF000000 | (uint32_t)((p)[0]) | ((uint32_t)((p)[1]) << 8) |               \
+   ((uint32_t)((p)[2]) << 16))
+
+#define GRAY16TOGRAY8(p, ns)                                                   \
+  (uint8_t)((((uint16_t)((p)[0])) | (((uint16_t)((p)[1])) << 8)) >> ns)
+
+#define BGR48TOARGB32(p)                                                       \
+  (0xFF000000 | (uint32_t)((p)[1]) | ((uint32_t)((p)[3]) << 8) |               \
+   ((uint32_t)((p)[5]) << 16))
+
+enum remixer {
+  RMX_CARIO24RGB,
```

Comment:
Cairo, not Cario.

Reply:
Corrected.

Reply:
I still see a number of instances of "Cario" in the diff.

Reply:
feel like this word is etched on my cortex.